application/common/components/adapters/WorkdayIdStore.php

```diff
@@ -153,4 +145,27 @@ public function getIdStoreName(): string
     {
         return 'Workday';
     }
+
+    public function generateGroupsLists(array &$users)
+    {
+        if ($this->groupsFields === null) {
+            $groupsFields = [
+                'company_ids',
+                'ou_tree',
+            ];
+        } else {
+            $groupsFields = explode(',', $this->groupsFields);
+        }
+
+        foreach ($users as $key => $user) {
+            $groups = [];
+            foreach ($groupsFields as $groupsField) {
+                if (strlen($user[$groupsField]) > 0) {
+                    $groupsSubList = explode(' ', $user[$groupsField ?? '']);
```

Comment:
I probably should have noticed this earlier, but why the use of `??` here?
`$user[$groupsField ?? '']`
Did you intend to do this instead?
`$user[$groupsField] ?? ''`